### PR TITLE
Updated search API to not return draft doi requests

### PR DESCRIPTION
**Pull request by RanjithTom**

### CODE REVIEW

File: search-commons/src/main/java/no/unit/nva/search/SearchClient.java [hunk 1]
```diff
@@ -95,7 +97,12 @@ private BoolQueryBuilder matchOneOfOrganizationIdsQuery(ViewingScope viewingScop
...
             queryBuilder.mustNot(matchPhraseQuery(ORGANIZATION_IDS, excludedOrganizationId.toString()));
         }
-        queryBuilder.mustNot(QueryBuilders.matchQuery(STATUS, APPROVED));
+        searchQueryToRemoveApprovedWorkListAndDraftPublications(queryBuilder);
         return queryBuilder;
     }
+
+    private void searchQueryToRemoveApprovedWorkListAndDraftPublications(BoolQueryBuilder queryBuilder) {
+        queryBuilder.mustNot(QueryBuilders.matchQuery(STATUS, APPROVED));
+        queryBuilder.mustNot(QueryBuilders.matchQuery(PUBLICATION_STATUS, DRAFT));
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/SearchClient.java:106]: is it not possible to say that we do not want PublicationStatus.DRADT only for DoiRequests?

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 2]
```diff
@@ -51,6 +53,30 @@ void constructorWithEnvironmentDefinedShouldCreateInstance() {
         assertNotNull(searchClient);
     }
 
+    @Test
+    void shouldSendRequestWithQueryToRemovePublicationStatusDraftWhenSearchingForResources() throws ApiGatewayException {
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:57]: shouldSendQueryThatFiltersOutDoiRequestsWithPublcationStatusDraftWhen...

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: search-commons/src/main/java/no/unit/nva/search/SearchClient.java (hunk 1 above)
Comment (RanjithTom) [reply search-commons/src/main/java/no/unit/nva/search/SearchClient.java:106]: BoolQueryBuilder have must, mustNot, should and filter clauses, I am not able to fine clauses to match our scenario like if it's DoiRequest Type then check for Publication Status else return.

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 3]
```diff
@@ -51,6 +53,30 @@ void constructorWithEnvironmentDefinedShouldCreateInstance() {
...
+        };
+
+        SearchClient searchClient = new SearchClient(restClientWrapper);
+        searchClient.findResourcesForOrganizationIds(generateSampleViewingScope(),
+                                                     DEFAULT_PAGE_SIZE,
+                                                     DEFAULT_PAGE_NO,
+                                                     ELASTICSEARCH_ENDPOINT_INDEX);
+        var sentRequest = sentRequestBuffer.get();
+        var query = sentRequest.source().query();
+        var expectedMustNotClause =   ((MatchQueryBuilder)((BoolQueryBuilder) query).mustNot().get(2)).value();
```
[the remark this answers is not in the document]
Comment (RanjithTom) [reply search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:76]: We have 3 MustNot clauses in the queryBuilder, 1. Excluded Organization Ids, 2. Approved worklist status and 3. Draft Publication Status. So to make sure we have set the Draft Pubilcation status in the query to elastic search

[the remark this answers is not in the document]
Comment (axthosarouris) [reply search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:76]: replace magic number with meaningful constant or variable.

File: search-commons/src/main/java/no/unit/nva/search/SearchClient.java [hunk 4]
```diff
@@ -95,7 +97,12 @@ private BoolQueryBuilder matchOneOfOrganizationIdsQuery(ViewingScope viewingScop
         for (URI excludedOrganizationId : viewingScope.getExcludedUnits()) {
             queryBuilder.mustNot(matchPhraseQuery(ORGANIZATION_IDS, excludedOrganizationId.toString()));
         }
-        queryBuilder.mustNot(QueryBuilders.matchQuery(STATUS, APPROVED));
+        searchQueryToRemoveApprovedWorkListAndDraftPublications(queryBuilder);
```
Comment (axthosarouris) [inline search-commons/src/main/java/no/unit/nva/search/SearchClient.java:100]: the method name does not make sense.  `excludeApprovedDoiRequestsAndDoiRequestsForDraftPublications`

File: search-commons/src/main/java/no/unit/nva/search/SearchClient.java (hunk 1 above)
Comment (axthosarouris) [reply search-commons/src/main/java/no/unit/nva/search/SearchClient.java:106]: Have you tried combining multiple boolean queries? https://stackoverflow.com/questions/28538760/elasticsearch-bool-query-combine-must-with-or

Comment (RanjithTom) [reply search-commons/src/main/java/no/unit/nva/search/SearchClient.java:106]: Tried with different combinations of boolean queries but not getting the exact result. For the time being will forward with current implementation ie. Look for publication.status = DRAFT.

Will create one bug task in Jira to fix this in future. And the fix needed : Check for publication.status only if it is a DoiRequest Type

Comment (RanjithTom) [reply search-commons/src/main/java/no/unit/nva/search/SearchClient.java:106]: As suggested created one bug in Jira

https://unit.atlassian.net/browse/NP-9090

File: search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java [hunk 5]
```diff
@@ -51,6 +53,74 @@ void constructorWithEnvironmentDefinedShouldCreateInstance() {
...
+                                                     ELASTICSEARCH_ENDPOINT_INDEX);
+        var sentRequest = sentRequestBuffer.get();
+        var query = sentRequest.source().query();
+        var doiRequestTypeClauseIndexInQuery = 1;
+        var doiRequestsQueryBuilder = ((BoolQueryBuilder) ((BoolQueryBuilder) query).should()
+            .get(doiRequestTypeClauseIndexInQuery));
+        var documentTypeIndexInQueryBuilderMustClause = 0;
+        var expectedDocumentTypeInMustClause = ((MatchQueryBuilder) doiRequestsQueryBuilder.must()
+            .get(documentTypeIndexInQueryBuilderMustClause)).value();
+        assertThat(expectedDocumentTypeInMustClause, is(equalTo("DoiRequest")));
```
Comment (axthosarouris) [inline search-commons/src/test/java/no/unit/nva/search/SearchClientTest.java:83]: replace with the constant DOI_REQUEST

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**